pypeit/core/flat.py

```diff
@@ -19,8 +19,353 @@
 from pypeit.core import parse
 from pypeit.core import qa
 from pypeit.core import pca
+from pypeit.core import pixels
 
 from pypeit import debugger
+from pypeit import utils
+from pypeit.core import pydl
+from matplotlib import pyplot as plt
+import scipy
+
+
+def fit_flat(flat, mstilts, thismask, slit_left, slit_righ, inmask = None,spec_samp_fine = 1.2,  spec_samp_coarse = 50.0,
```

Comment:
There seem to be a lot of hard-coded numbers.
 - How flexible is this to different instruments?
 - Can we remove any hard-coded numbers and make them function arguments with default values instead?

Reply:
This will be mapped to parsets soon allowing people to modify  these numbers from the defaults if necessary. I've tested on many instruments. The numbers are chosen in a way that they are in pixel units and are basically "scale" free, i.e. they should adapt to any resolution etc.